src/NuGet.Core/NuGet.Packaging/Licenses/NuGetLicenseException.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+
+namespace NuGet.Packaging.Licenses
+{
+    /// <summary>
+    /// NuGet's internal representation of a license exception identifier. 
+    /// </summary>
+    public class NuGetLicenseException
```

Comment:
It's not part of a nuget license. 
They are separate. 

LICENSE WITH EXCEPTION is a complex expression by definition. 
Any license X exception combination si acceptable.

You cannot have an exception without a license ID, hence why NuGetLicenseException does not inherit NuGetLicenseExpression.

Reply:
`You cannot have an exception without a license ID, hence why NuGetLicenseException does not inherit NuGetLicenseExpression.`

That's why I dont understand why there is a class with only exception identifier when it can never happens without a license id... May be we can discuss it offline to get clarity.

Reply:
Because it's more extensible for the future. We have an object that represents a exception whatever that may be. Future additions to exceptions (maybe we care about OSI approved licenses/exceptions for example for some feature), would be backwards compatible for code using the parser because they won't change the types of what a WithOperator represents.